pkg/services/ngalert/schedule/schedule.go

```diff
@@ -511,7 +511,9 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 		case <-grafanaCtx.Done():
 			// We need to use context.Background() here because grafanaCtx is being canceled, and we need a valid context
 			// in clearState function to do DB operations.
-			clearState(context.Background(), grafanaCtx.Err())
+			ctx, cancelFunc := context.WithTimeout(context.Background(), 5 * time.Second)
+			defer cancelFunc()
```

Comment:
Apologies, we might need to remove the `cancelFunc` I think as annotations are done async, the context will be canceled when this function returns, not when the timeout has expired.